lib/src/dummies.dart

```diff
@@ -16,15 +16,19 @@ import 'dart:async';
 import 'dart:collection';
 import 'dart:typed_data';
 
-import 'mock.dart' show FakeFunctionUsedError;
+import 'mock.dart' show FakeFunctionUsedError, PrettyString;
 import 'platform_dummies_js.dart'
     if (dart.library.io) 'platform_dummies_vm.dart';
 
 // TODO(yanok): try to change these to _unreasonable_ values, for example,
 // String could totally contain an explanation.
 const int _dummyInt = 0;
 const double _dummyDouble = 0.0;
-const String _dummyString = '';
+
+// Create a dummy String with info on why it was created.
+String _dummyString(Object o, Invocation i) => Uri.encodeComponent(
+    'Dummy String created while calling ${i.toPrettyString()} on $o.'
+        .replaceAll(' ', '_'));
```

Comment:
Why do we need the `_` replacement?

Reply:
We don't technically. But I find 'Dummy_String_created_while_calling_Foo' more readable than 'Dummy%20String%20created%20while%20calling%20Foo'

Reply:
Ah, why do we need to encode it as a URI?

Reply:
Hmm, if we could automate it I'd have a preference for adding `skip` in impacted tests or similar and asking owning teams to clean it up. Leaking these usage details backwards into the implementation has me worried.

I haven't been following any details of getting this landed though, so I won't push back further if we need this for google3.

Reply:
Yeah, I agree that's not ideal, but I like to think about it as a gradual process: we started with the least breaking thing, which is an empty string and now we start to make it more and more breaking: at this point it still has no whitespace and it's URI component encoded, but we could remove this at a later stage.